### PR TITLE
Move `DetectWSL` function into util package

### PR DESCRIPTION
**Pull request by gsquared94**
Moved `DetectWSL` function into util package. This is a prereq for #4701

### CODE REVIEW

File: pkg/skaffold/docker/client.go [hunk 1]
```diff
@@ -221,7 +206,7 @@ func getMinikubeDockerEnv(minikubeProfile string) (map[string]string, error) {
 		env[kv[0]] = kv[1]
 	}
 
-	if found, _ := detectWsl(); found {
+	if found, _ := util.DetectWSL(); found {
 		cmd := exec.Command("wslpath", env["DOCKER_CERT_PATH"])
```
Comment (briandealwis) [inline pkg/skaffold/docker/client.go:210]: I know this isn't part of your change but could you just add a comment here to the effect of:
```
// rewrite Unix path to Windows
```

(at least, I'm assuming Minikube in WSL reports a unix-style path?)

File: pkg/skaffold/util/wsl.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+func DetectWSL() (bool, error) {
+	if _, err := os.Stat("/proc/version"); err == nil {
+		b, err := ioutil.ReadFile("/proc/version")
+		if err != nil {
+			return false, fmt.Errorf("read /proc/version: %w", err)
+		}
+
+		str := strings.ToLower(string(b))
+		if strings.Contains(str, "microsoft") {
```
Comment (briandealwis) [inline pkg/skaffold/util/wsl.go:34]: Maybe a comment here
```
// Microsoft changed the case between WSL1 and 2
```

File: pkg/skaffold/docker/client.go (hunk 1 above)
Comment (gsquared94) [reply pkg/skaffold/docker/client.go:211]: done

File: pkg/skaffold/util/wsl.go (hunk 2 above)
Comment (gsquared94) [reply pkg/skaffold/util/wsl.go:36]: done

**Pull request merged**